qiskit_experiments/curve_analysis/utils.py

```diff
@@ -115,6 +115,7 @@ def convert_lmfit_result(
     models: List[lmfit.Model],
     xdata: np.ndarray,
     ydata: np.ndarray,
+    residuals: Optional[np.ndarray],
```

Comment:
1. `Optional` is not needed when there's no default value.
2. The documented type of `residuals` here is `Optional[np.ndarray]`, which is a shorthand of `Union[np.ndarray, None`]. However if I go back to the call to `convert_lmfit_result`, in `CurveAnalysis._run_curve_fit`, then the real type appears to be `Union[np.array, List]`.

Reply:
You are right. I changed it to be:
```
residuals: Union[np.array, List],
```